Mathlib/MeasureTheory/Group/ModularCharacter.lean

```diff
@@ -0,0 +1,135 @@
+/-
+Copyright (c) 2025 Noam Atar. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Noam Atar
+-/
+import Mathlib.MeasureTheory.Function.LocallyIntegrable
+import Mathlib.MeasureTheory.Group.Integral
+import Mathlib.MeasureTheory.Group.Measure
+import Mathlib.Topology.Metrizable.Urysohn
+import Mathlib.MeasureTheory.Measure.Haar.Unique
+
+
+/-!
+# Modular character of a locally compact group
+
+On a locally compact group, there is a natural homomorphism `G → ℝ≥0*`, which for `g : G` gives the
+value `μ (· * g⁻¹) / μ`, where `μ` is an (inner regular) Haar measure. This file defines this
+homomorphism, called the modular character, and shows that it is independent of the chosen Haar
+measure.
+
+TODO: Show that the character is continuous.
+
+## Main Declarations
+
+* `modularCharacterFun`: Define the modular character associated with this measure, which is
+  `g ↦ haarScalarFactor (map (· * g) ν) ν`, where ν is the haar measure given by (the noncomputable)
+  `MeasureTheory.Measure.haar`. The result that this does not depend on the measure chosen is
+  `modularCharacterFun_eq_haarScalarFactor`.
+* `modularCharacter`: The homomorphism G →* ℝ≥0 whose toFun is `modularCharacterFun`.
+-/
+
+open MeasureTheory
+open scoped NNReal
+
+namespace MeasureTheory
+
+namespace Measure
+
+variable {G : Type*} [TopologicalSpace G] [Group G] [IsTopologicalGroup G]
+  [MeasurableSpace G] [BorelSpace G] [LocallyCompactSpace G]
+
+/-- The modular character as a map is `g ↦ haarScalarFactor (map (· * g) ν) ν`, where `ν` is
+  the haar measure given by (the noncomputable) `MeasureTheory.Measure.haar`.
+
+  See also `modularCharacter` that defines the map as a homomorphism. -/
+@[to_additive addModularCharacterFun "Definition of the addModularCharacterFun map as
+  `addHaarScalarFactor (map (· * g) ν) ν`, where `ν` is
+  the haar measure given by (the noncomputable) MeasureTheory.Measure.haar.
+
+  Also `modularCharacter` defines the map as a homomorphism."]
```

Comment:
The phrasing of the additive version should match more closely the multiplicative one (and replace `*` with `+`)